Description: Don't create the top panel.
Author: Marc Deslauriers <marc.deslauriers@canonical.com>
Forwarded: no, Ubuntu-specific

Index: gnome-screensaver/src/gs-window-x11.c
===================================================================
--- gnome-screensaver.orig/src/gs-window-x11.c	2012-02-11 02:24:37.000000000 +0100
+++ gnome-screensaver/src/gs-window-x11.c	2012-03-07 15:13:33.797846680 +0100
@@ -2335,10 +2335,18 @@
         gtk_widget_show (window->priv->vbox);
         gtk_container_add (GTK_CONTAINER (window), window->priv->vbox);
 
-        window->priv->panel = gtk_box_new (GTK_ORIENTATION_HORIZONTAL, 12);
-        gtk_widget_show (window->priv->panel);
-        gtk_box_pack_start (GTK_BOX (window->priv->vbox), window->priv->panel, FALSE, FALSE, 0);
-        create_panel (window);
+        /* Only display top bar if we're not running under Unity */
+        if (!g_getenv ("XDG_CURRENT_DESKTOP") ||
+            strcmp (g_getenv ("XDG_CURRENT_DESKTOP"), "Unity") != 0) {
+                window->priv->panel = gtk_box_new (GTK_ORIENTATION_HORIZONTAL, 12);
+                gtk_widget_show (window->priv->panel);
+                gtk_box_pack_start (GTK_BOX (window->priv->vbox), window->priv->panel, FALSE, FALSE, 0);
+                create_panel (window);
+
+                window->priv->clock_tracker = g_object_new (GNOME_TYPE_WALL_CLOCK, NULL);
+                g_signal_connect (window->priv->clock_tracker, "notify::clock", G_CALLBACK (on_clock_changed), window);
+                update_clock (window);
+        }
 
         window->priv->drawing_area = gtk_drawing_area_new ();
         gtk_widget_show (window->priv->drawing_area);
@@ -2346,10 +2354,6 @@
         gtk_box_pack_start (GTK_BOX (window->priv->vbox), window->priv->drawing_area, TRUE, TRUE, 0);
         create_info_bar (window);
 
-        window->priv->clock_tracker = g_object_new (GNOME_TYPE_WALL_CLOCK, NULL);
-        g_signal_connect (window->priv->clock_tracker, "notify::clock", G_CALLBACK (on_clock_changed), window);
-        update_clock (window);
-
         force_no_pixmap_background (window->priv->drawing_area);
 }
 
