Description: Run the autostart file for Unity, too.
Author: Martin Pitt <martin.pitt@ubuntu.com>
Forwarded: No, Unity/Ubuntu specific

Index: gnome-screensaver-3.2.0/src/gnome-screensaver.desktop.in
===================================================================
--- gnome-screensaver-3.2.0.orig/src/gnome-screensaver.desktop.in	2012-01-05 12:46:55.631177542 +0100
+++ gnome-screensaver-3.2.0/src/gnome-screensaver.desktop.in	2012-01-05 12:47:30.223179217 +0100
@@ -4,7 +4,7 @@
 _Comment=Launch screensaver and locker program
 Icon=preferences-desktop-screensaver
 Exec=gnome-screensaver
-OnlyShowIn=GNOME;
+OnlyShowIn=GNOME;Unity;
 NoDisplay=true
 X-GNOME-Autostart-Delay=20
 X-GNOME-Autostart-Phase=Application
Index: gnome-screensaver-3.2.0/src/gnome-screensaver.desktop.in.in
===================================================================
--- gnome-screensaver-3.2.0.orig/src/gnome-screensaver.desktop.in.in	2012-01-05 12:46:55.631177542 +0100
+++ gnome-screensaver-3.2.0/src/gnome-screensaver.desktop.in.in	2012-01-05 12:47:28.195179119 +0100
@@ -4,7 +4,7 @@
 _Comment=Launch screensaver and locker program
 Icon=preferences-desktop-screensaver
 Exec=gnome-screensaver
-OnlyShowIn=GNOME;
+OnlyShowIn=GNOME;Unity;
 NoDisplay=true
 X-GNOME-Autostart-Delay=20
 X-GNOME-Autostart-Phase=Application
